File: example/ios/Runner/ExampleClass.swift
@objc protocol ExampleClass {
  var name: String { get }
}

class Apple: ExampleClass {
  var name: String = "Apple"
}

class Orange: ExampleClass {
  var name: String = "Orange"
}

class Banana: ExampleClass {
  var name: String = "Banana"
}

class NotFruit: ExampleClass {
  var name: String
  init(_ name: String) {
    self.name = name
  }
}

File: example/android/app/src/main/kotlin/br/dig/dev/flutter/flutter_reference_example/ExampleClass.kt
package br.dig.dev.flutter.flutter_reference_example

sealed interface ExampleClass {
    val name: String
}

object Apple : ExampleClass {
    override val name: String
        get() = "Apple"
}

object Orange : ExampleClass {
    override val name: String
        get() = "Orange"
}

object Banana : ExampleClass {
    override val name: String
        get() = "Banana"
}

class NotFruit(override val name: String) : ExampleClass